src/mapping/RadialBasisFctMapping.hpp

```diff
@@ -122,84 +126,137 @@ void RadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::computeMapping()
+      com::CommunicateMesh(utils::MasterSlave::_communication).receiveMesh(slaveOutMesh, rankSlave);
+      globalOutMesh.addMesh(slaveOutMesh);
     }
-  }
 
-  // Fill _matrixA with values
-  i = 0;
-  for (const mesh::Vertex &iVertex : outMesh->vertices()) {
-    int j = 0;
-    for (const mesh::Vertex &jVertex : inMesh->vertices()) {
-      difference = iVertex.getCoords();
-      difference -= jVertex.getCoords();
-      _matrixA(i, j) = _basisFunction.evaluate(reduceVector(difference).norm());
-      j++;
+    int inputSize  = (int) globalInMesh.vertices().size();
+    int outputSize = (int) globalOutMesh.vertices().size();
+    PRECICE_DEBUG("Size of input mesh: " << inputSize);
+    PRECICE_DEBUG("Size of output mesh: " << outputSize);
+
+    int deadDimensions = 0;
+    for (int d = 0; d < dimensions; d++) {
+      if (_deadAxis[d])
+        deadDimensions += 1;
     }
-    _matrixA(i, inputSize) = 1.0;
-    for (int dim = 0; dim < dimensions - deadDimensions; dim++) {
-      _matrixA(i, inputSize + 1 + dim) = reduceVector(iVertex.getCoords())[dim];
+    int polyparams = 1 + dimensions - deadDimensions;
+    PRECICE_ASSERT(inputSize >= 1 + polyparams, inputSize);
+    int n = inputSize + polyparams; // Add linear polynom degrees
+
+    // LU Decomposition
+    Eigen::MatrixXd matrixCLU(n, n);
+    matrixCLU.setZero();
+    _matrixA = Eigen::MatrixXd(outputSize, n);
+    _matrixA.setZero();
+    // Fill upper right part (due to symmetry) of _matrixCLU with values
+    Eigen::VectorXd difference(dimensions);
+    for (int i = 0; i < inputSize; ++i) {
+      for (int j = i; j < inputSize; ++j) {
+        difference = globalInMesh.vertices()[i].getCoords();
+        difference -= globalInMesh.vertices()[j].getCoords();
+        matrixCLU(i, j) = _basisFunction.evaluate(reduceVector(difference).norm());
+      }
+      matrixCLU(i, inputSize) = 1.0;
+      for (int dim = 0; dim < dimensions - deadDimensions; dim++) {
+        matrixCLU(i, inputSize + 1 + dim) = reduceVector(globalInMesh.vertices()[i].getCoords())[dim];
+      }
     }
-    i++;
-  }
-
-  _qr = matrixCLU.colPivHouseholderQr();
-  if (not _qr.isInvertible()) {
-    PRECICE_ERROR("RBF interpolation matrix is not invertible! "
-                  "Try to fix axis-aligned mapping setups by marking perpendicular axes as dead.");
+    // Copy values of upper right part of C to lower left part
+    for (int i = 0; i < n; i++) {
+      for (int j = i + 1; j < n; j++) {
+        matrixCLU(j, i) = matrixCLU(i, j);
+      }
+    }
+    // Fill _matrixA with values
+    for (int i = 0; i < outputSize; ++i) {
+      for (int j = 0; j < inputSize; ++j) {
+        difference = globalOutMesh.vertices()[i].getCoords();
+        difference -= globalInMesh.vertices()[j].getCoords();
+        _matrixA(i, j) = _basisFunction.evaluate(reduceVector(difference).norm());
+      }
+      _matrixA(i, inputSize) = 1.0;
```

Comment:
I investigated the matrix and actually we not set the entire column at the end. So one option that I only find is block operations and I think it would be an overkill for this process. However, I moved it to one outer loop, at least it won't be set twice.